main_test.go

```diff
@@ -0,0 +1,145 @@
+//go:build cassandra || integration
+// +build cassandra integration
+
+package gocql
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+func TestMain(m *testing.M) {
+	ctx := context.Background()
+
+	flag.Parse()
+
+	cassandraVersion := flagCassVersion.String()[1:]
+
+	jvmOpts := "-Dcassandra.test.fail_writes_ks=test -Dcassandra.custom_query_handler_class=org.apache.cassandra.cql3.CustomPayloadMirroringQueryHandler"
+	if *clusterSize == 1 {
+		// speeds up the creation of a single-node cluster.
+		jvmOpts += " -Dcassandra.initial_token=0 -Dcassandra.skip_wait_for_gossip_to_settle=0"
+	}
+
+	env := map[string]string{
+		"JVM_OPTS":                  jvmOpts,
+		"CASSANDRA_SEEDS":           "cassandra1",
+		"CASSANDRA_DC":              "datacenter1",
+		"HEAP_NEWSIZE":              "100M",
+		"MAX_HEAP_SIZE":             "256M",
+		"CASSANDRA_RACK":            "rack1",
+		"CASSANDRA_ENDPOINT_SNITCH": "GossipingPropertyFileSnitch",
+		"CASS_VERSION":              cassandraVersion,
+	}
+
+	if *flagRunAuthTest {
+		env["AUTH_TEST"] = "true"
+	}
+
+	networkRequest := testcontainers.GenericNetworkRequest{
+		NetworkRequest: testcontainers.NetworkRequest{
+			Name: "cassandra",
+		},
+	}
+	cassandraNetwork, err := testcontainers.GenericNetwork(ctx, networkRequest)
+	if err != nil {
+		log.Fatalf("Failed to create network: %s", err)
+	}
+	defer cassandraNetwork.Remove(ctx)
+
+	// Function to create a Cassandra container (node)
+	createCassandraContainer := func(number int) (string, error) {
```

Comment:
I'm not a Cassandra expert, so you probably want to take a look and customise the existing cassandra Go module with your code: https://golang.testcontainers.org/modules/cassandra/ (even contributing to the module with updates would be great!)